### ISSUE TITLE
Community Suggestion - RC Budget Witness Parameters

### ISSUE THREAD

**Issue by TimCliff** [steemit/steem#2991]
The current implementation of RC budgets requires all witnesses to use the same budgets, otherwise there will be inconsistencies of behavior between nodes. Also, each time a budget is changed, it requires a code change and replay.

It seems that a more optimal way of handling changes in the RC budgets would be for witnesses to vote on each budget via a witness parameter, and have the blockchain use the median value as the current budget.

- [ ] Create new witness parameters for the RC budgets.
- [ ] Update the RC budgets to be based on the median value across the witnesses.
- [ ] Update the witness parameter documentation to include the new parameters.

**Comment by Smittyvb** [comment]
Just to be clear, would RCs still be implemented as a soft fork?

**Comment by TimCliff** [comment]
The witness parameter values would need to become part of consensus (afaik) but their enforcement through the RC plugin could still be done via non-consensus / soft-fork.

**Comment by iamsmooth** [comment]
I don't think this should be witness settable. One of the factors that go into RC limits is ability to run nodes which implicates all of those running nodes in the ecosystem, including those whose economic influence may far exceed their stake (such as large exchanges). Require all members of the ecosystem to buy into these limits via a node upgrade is more sensible than allowing witnesses to arbitrarily adjust (which in effect puts the decision mostly in the hands of the largest stakeholders).
